app/controlplane/api/workflowcontract/v1/crafting_schema_validations.go

```diff
@@ -68,3 +74,95 @@ func (schema *CraftingSchema) ValidateUniqueMaterialName() error {
 
 	return nil
 }
+
+func (schema *CraftingSchema) ValidatePolicyAttachments() error {
+	attachments := append(schema.GetPolicies().GetAttestation(), schema.GetPolicies().GetMaterials()...)
+
+	for _, att := range attachments {
+		if err := ValidatePolicyAttachmentRef(att.GetRef()); err != nil {
+			return fmt.Errorf("invalid reference %q: %w", att.GetRef(), err)
+		}
+	}
+
+	return nil
+}
+
+func ValidatePolicyAttachmentRef(ref string) error {
+	if ref == "" {
+		return fmt.Errorf("reference is empty")
+	}
+
+	// validate the optional digest format
+	parts := strings.SplitN(ref, "@", 2)
+	if len(parts) > 1 {
+		rawDigest := parts[1]
+		if _, err := cr_v1.NewHash(rawDigest); err != nil {
+			return fmt.Errorf("invalid digest, want policy-ref@sha256:[hex]: %w", err)
+		}
+
+		// remove it @sha256: suffix
+		ref = strings.TrimSuffix(ref, fmt.Sprintf("@%s", parts[1]))
+	}
+
+	u, err := url.Parse(ref)
+	if err != nil {
+		return fmt.Errorf("invalid reference: %w", err)
+	}
+
+	switch u.Scheme {
+	case "http", "https", "file":
+		if u.Path == "" {
+			return fmt.Errorf("path is empty")
+		} else if filepath.Ext(u.Path) == "" {
```

Comment:
We are checking any extension here right? We could potentially type `/deadbeef.foo` no?
Shall we tighten it to only `.yaml|yml`?

Reply:
yeah, maybe, tbh I am not sure. I left it open since we might support, and I think we do support json too. This check is more about making sure you are pointing to an actual file.